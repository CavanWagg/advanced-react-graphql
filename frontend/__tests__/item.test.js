import ItemComponent from '../components/Item';
import { shallow, mount } from 'enzyme';
import toJSON from 'enzyme-to-json';

const fakeItem = {
  id: 'asdfjkl123',
  title: 'A cool Item',
  price: 10000,
  description: 'Sickkkkkk',
  image: 'happy.jpg',
  largeImage: 'large.jpg',
};

describe('<Item/>', () => {
  it('renders and matches the snapshot', () => {
    const wrapper = shallow(<ItemComponent item={fakeItem} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  })

  // it('renders the image properly', () => {
  //   const wrapper = shallow(<ItemComponent item={fakeItem} />);
  //   const img = wrapper.find('img');
  //   expect(img.props().src).toBe(fakeItem.image);
  //   expect(img.props().alt).toBe(fakeItem.title);
  // })
  // it('renders the pricetag and title', () => {
  //   const wrapper = shallow(<ItemComponent item={fakeItem} />);
  //   const PriceTag = wrapper.find('PriceTag');
  //   expect(PriceTag.children().text()).toBe('$100');
  //   const Title = (wrapper.find('Title a'));
  //   expect(Title.text()).toBe(fakeItem.title);
  // });

  // it('renders out the buttons properly', () => {
  //   const wrapper = shallow(<ItemComponent item={fakeItem} />);
  //   const buttonList = wrapper.find('.buttonList');
  //   expect(buttonList.children()).toHaveLength(3);
  //   expect(buttonList.find('Link')).toHaveLength(1);
  //   expect(buttonList.find('Link')).toBeTruthy();
  //   expect(buttonList.find('AddToCart')).toBeTruthy();
  //   expect(buttonList.find('DeleteItem')).toBeTruthy();
  // })
});

