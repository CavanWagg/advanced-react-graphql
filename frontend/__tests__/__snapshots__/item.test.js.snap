// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Item/> renders and matches the snapshot 1`] = `
<ItemStyles__Item>
  <img
    alt="A cool Item"
    src="happy.jpg"
  />
  <Title>
    <Link
      href={
        Object {
          "pathname": "/item",
          "query": Object {
            "id": "asdfjkl123",
          },
        }
      }
    >
      <a>
        A cool Item
      </a>
    </Link>
  </Title>
  <PriceTag>
    $100
  </PriceTag>
  <p>
    Sickkkkkk
  </p>
  <div
    className="buttonList"
  >
    <Link
      href={
        Object {
          "pathname": "update",
          "query": Object {
            "id": "asdfjkl123",
          },
        }
      }
    >
      <a>
        Edit ✏️
      </a>
    </Link>
    <AddToCart
      id="asdfjkl123"
    />
    <DeleteItem
      id="asdfjkl123"
    >
      Delete This Item
    </DeleteItem>
  </div>
</ItemStyles__Item>
`;
